### DISCUSSION TITLE
More inpaint tools

### DISCUSSION THREAD

**Discussion by Xposure345** [lllyasviel/Fooocus#1477] [Ideas]
It could be great to have a fuzzy select tool, like in photoshop to try and select what you want automatically. I was also thinking just a line tool for sharp edges, and a fill bucket tool to fill in selections. Maybe even circle selection would be nice. Subject selection? Face selection? I think more drawing tools for inpaint would be great. Thanks for your time!

**Comment by xhoxye** [comment]
You want to turn it into PS

**Comment by Xposure345** [comment]
Yes, that’s what I said because I suggested tools from PS I meant turn the whole app to PS, you are brilliant.
Sent from my iPhone

On Dec 17, 2023, at 5:45 PM, xhoxye ***@***.***> wrote:

﻿

You want to turn it into PS

**Comment by xhoxye** [comment]
But this is not possible, because it would conflict with the philosophy of Fooocus.
you don't need complicated tools, you can now zoom in on the brush and zoom out on the canvas.

**Reply by Alexej-Larionov** [reply]
I think that color picker would be the best way to improve masking in inpaint/outpaint tool and make it QoL. Also I guess, it's a simple tool to use and relatively easy to implement (it seems).

**Comment by mkfoto** [comment]
Would be great to have different inpaint sections with separate prompts. Like blue color - blue input prompt, red color - red input prompt, etc.. Would made things more precise and faster.. :)
